src/context/history-context/history-context.js

```diff
@@ -0,0 +1,126 @@
+import { useContext, useState, createContext } from "react";
+import { useEffect } from "react";
+import axios from "axios";
+import { useToast } from "../toast-context/toast-context";
+
+const HistoryContext = createContext();
+
+const useHistory = () => useContext(HistoryContext);
+
+function HistoryProvider({ children }) {
+  const encodedToken = localStorage.getItem("smashTubeToken");
+  const [history, setHistory] = useState([]);
+  const { dispatch } = useToast();
+
+  useEffect(() => {
+    (async function () {
+      if (encodedToken) {
+        try {
+          const historyResponse = await axios.get("/api/user/history", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
```

Comment:
I would suggest that try to do it like this. ` useEffect(()=> {
if(encodedToken){
then your asncy func
}
},[dep])`

In your case you are running the function and then checking whether the encoded token is there or not.

Reply:
Okay will check